spdk/src/io_device.rs

```diff
@@ -0,0 +1,103 @@
+use std::os::raw::c_void;
```

Comment:
We already have implementation of SPDK I/O device in Rust:
```
src/core/io_device.rs
```
There is also a very important feature which is missing: I/O channel enumeration, which is widely used in Mayastor.
Our current Rust implementation of I/O channel contains it, so you can take it as an example.

Reply:
I used the Mayastor version as an example. I will add missing features a bit later. First, I needed to figure out the fundamental functionality: who owns who, how SPDK interacts with Rust code, etc.